tests/models/faraday/gaussian_mixture/test_gmm_inputs.py

```diff
@@ -22,17 +23,18 @@ class TestGMMDataPreparation:
 
     unweighted_batch = TrainingData(kwh=kwh, features=features)
```

Comment:
In the line above:
```Python
batch = TrainingData(kwh=kwh, features=features, weights=weights)
```

This shouldn't be allowed. Not sure why mypy has not detected this (see https://peps.python.org/pep-0589/).

The proper way to do this is:

```python
class TrainingDataWithWeights(TrainingData):
    weights: torch.Tensor
```

See this commit:
https://github.com/OpenSynth-energy/OpenSynth/blob/42624dcbb5a612c6058fcfdc55d08c4c8fdd5617/tests/models/faraday/gaussian_mixture/test_gmm_inputs.py

Reply:
Updated.